### PR TITLE
build: Fix duplicate symbol error of string.h

### LINKED ISSUE #9367: Compilation fails on RockyLinux 8 arm64 arch
## Bug Report

Please answer these questions before submitting your issue. Thanks!

```
FAILED: libs/libarm-optimized-routines/CMakeFiles/aor.dir/aor.cpp.o 
/usr/bin/clang++ -DSERVERLESS_PROXY=0 -DTIFLASH_COMPILER_VPCLMULQDQ_SUPPORT=0 -DTIFLASH_ENABLE_ASIMD_SUPPORT=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/ws/contrib/arm-optimized-routines/string/include -Wthread-safety -flto=thin -fvisibility=hidden -fvisibility-inlines-hidden -fwhole-program-vtables -fsplit-lto-unit  -pipe -march=armv8+crc+simd  -fno-omit-frame-pointer  -Wall -Wno-unused-command-line-argument  -Wnon-virtual-dtor  -stdlib=libc++ -Werror -fomit-frame-pointer -O2 -g -DNDEBUG -O3  -std=gnu++2b -MD -MT libs/libarm-optimized-routines/CMakeFiles/aor.dir/aor.cpp.o -MF libs/libarm-optimized-routines/CMakeFiles/aor.dir/aor.cpp.o.d -o libs/libarm-optimized-routines/CMakeFiles/aor.dir/aor.cpp.o -c /ws/libs/libarm-optimized-routines/aor.cpp
/ws/libs/libarm-optimized-routines/aor.cpp:35:58: error: 'memchr' is missing exception specification 'throw()'
   35 | extern "C" __attribute__((visibility("default"))) void * memchr(const void * src, int c, size_t size)
      |                                                          ^                                           
      |                                                                                                       throw()
/usr/include/string.h:84:1: note: previous declaration is here
   84 | memchr (const void *__s, int __c, size_t __n) __THROW
      | ^
/ws/libs/libarm-optimized-routines/aor.cpp:40:58: error: 'memrchr' is missing exception specification 'throw()'
   40 | extern "C" __attribute__((visibility("default"))) void * memrchr(const void * src, int c, size_t size)
      |                                                          ^                                            
      |                                                                                                        throw()
/usr/include/string.h:112:26: note: previous declaration is here
  112 | extern "C++" const void *memrchr (const void *__s, int __c, size_t __n)
      |                          ^
/ws/libs/libarm-optimized-routines/aor.cpp:69:65: error: 'strchr' is missing exception specification 'throw()'
   69 | extern "C" __attribute__((visibility("default"))) inline char * strchr(const char * src_, int c)
      |                                                                 ^                               
      |                                                                                                  throw()
/usr/include/string.h:219:1: note: previous declaration is here
  219 | strchr (const char *__s, int __c) __THROW
      | ^
/ws/libs/libarm-optimized-routines/aor.cpp:74:65: error: 'strrchr' is missing exception specification 'throw()'
   74 | extern "C" __attribute__((visibility("default"))) inline char * strrchr(const char * src_, int c)
      |                                                                 ^                                
      |                                                                                                   throw()
/usr/include/string.h:246:1: note: previous declaration is here
  246 | strrchr (const char *__s, int __c) __THROW
      | ^
/ws/libs/libarm-optimized-routines/aor.cpp:79:65: error: 'strchrnul' is missing exception specification 'throw()'
   79 | extern "C" __attribute__((visibility("default"))) inline char * strchrnul(const char * src_, int c)
      |                                                                 ^                                  
      |                                                                                                     throw()
/usr/include/string.h:263:26: note: previous declaration is here
  263 | extern "C++" const char *strchrnul (const char *__s, int __c)
      |                          ^
5 errors generated.
```

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #9367

Problem Summary: Compilation fails on RockyLinux 8 arm64 arch as the issue show

### What is changed and how it works?

```commit-message
Fix compilation fails on RockyLinux 8 arm64 arch
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: libs/libarm-optimized-routines/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-option(USE_ARM_OPTIMIZED_ROUTINES "Use ARM optimized routines" ON)
+# TODO Enabling this feature causes "is missing exception specification" error on rockylinux8 and arm64
```
Comment (Lloyd-Pottiger) [inline libs/libarm-optimized-routines/CMakeLists.txt:15]: Why not just remove those functions with issues?

Comment (CalvinNeo) [reply libs/libarm-optimized-routines/CMakeLists.txt:15]: Is it useful if we can fix somehow?

**Comment by breezewish** [comment]
Only some of the rarely used functions are marked as inline and cannot be overridden. How about using approaches like https://github.com/tidbcloud/tiflash-cse/pull/245/files#diff-0d98be940370fadc881e27b4abc31168c77c6f65a03a3ec365ef3dd131c1b46a and https://github.com/tidbcloud/tiflash-cse/pull/248/files?

**Comment by CalvinNeo** [comment]
> Only some of the rarely used functions are marked as inline and cannot be overridden. How about using approaches like https://github.com/tidbcloud/tiflash-cse/pull/245/files#diff-0d98be940370fadc881e27b4abc31168c77c6f65a03a3ec365ef3dd131c1b46a and https://github.com/tidbcloud/tiflash-cse/pull/248/files?

Yes, you are right. I have fixed according to cse.

**Review by breezewish — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**